rbac/api/utils.py

```
@@ -182,19 +182,46 @@ def reset_imported_tenants(query: str, limit: int, excluded: list[str]):
-                (tuple(excluded),),
-            )
-        result = cursor.rowcount
-
-    logger.info(f"Deleted {result} tenants.")
+    try:
+        with connection.cursor() as cursor:
+            if limit > 0:
+                subquery = f"SELECT id {query}"
+                cursor.execute(
+                    "WITH delete_batch AS (" + subquery + ") "
+                    "DELETE FROM api_tenant as t USING delete_batch as del where t.id = del.id",
```

Ruff (0.15.21)

[error] Possible SQL injection vector through string-based query construction

(S608)